### PR TITLE
Add JPEG EXIF parser

### PR DESCRIPTION
**Pull request by szkarpinski**
## Category:

**New feature** (*non-breaking change which adds functionality*)

## Description:

Extends the imgcodec JPEG parser with ability to parse EXIFs orientation data.

Note: requires images from https://github.com/NVIDIA/DALI_extra/pull/87

### Affected modules and functionalities:
Imgcodec's JPEG parser can now handle EXIF data

### Tests:

- [ ] Existing tests apply
- [x] New tests added
  - [ ] Python tests
  - [x] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: DALI-2858

### CODE REVIEW

File: dali/imgcodec/parsers/jpeg.cc [hunk 1]
```diff
@@ -66,6 +72,18 @@ ImageInfo JpegParser::GetInfo(ImageSource *encoded) const {
       auto nchannels = stream->ReadOne<uint8_t>();
       info.shape = {height, width, nchannels};
       read_shape = true;
+    } else if (marker == app1_marker) {
+      DALI_ENFORCE(stream->ReadOne<jpeg_exif_header_t>() == exif_header, "Invalid EXIF header");
```
Comment (jantonguirao) [inline dali/imgcodec/parsers/jpeg.cc:76]: question: Is this supposed to be the only valid header within that app1 marker?

**Review by jantonguirao — APPROVED** [review]

File: dali/imgcodec/parsers/jpeg.cc (hunk 1 above)
Comment (szkarpinski) [reply dali/imgcodec/parsers/jpeg.cc:76]: Oh, nice catch! I'll fix that in a moment

Comment (szkarpinski) [reply dali/imgcodec/parsers/jpeg.cc]: Fixed in https://github.com/NVIDIA/DALI/pull/4073/commits/f2147abf0883d45684a581436373491ad4d9e6e4

**Review by awolant — COMMENTED** [review]
As far as I understand this PR should update DALI_EXTRA_VERSION, right?

**Comment by szkarpinski** [comment]
> As far as I understand this PR should update DALI_EXTRA_VERSION, right?

That's right, thanks for reminder. I'll run the CI manually in a moment to test if nothing fails (as it did recently :P), and then I'll merge https://github.com/NVIDIA/DALI_extra/pull/87 and update DALI_EXTRA_VERSION

**Comment by szkarpinski** [comment]
!build

**Review by awolant — APPROVED** [review]

**Pull request merged**